core/src/main/java/brooklyn/catalog/internal/CatalogDo.java

```diff
@@ -254,8 +270,10 @@ CatalogDo addCatalog(CatalogDto child) {
     public synchronized void addToClasspath(String ...urls) {
         if (dto.classpath == null)
             dto.classpath = new CatalogClasspathDto();
-        for (String url: urls)
-            dto.classpath.addEntry(url);
+        for (String url: urls) {
+            if (url!=null)
```

Comment:
the only non-test use is when we are registering declared osgi bundles.  a pre-installed bundle might have a null url.  we could filter this before calling but it is more work.

agree it's worth warning about nulls in situations where programming errors could crop up, but it seems unlikely here and the investment not worth it.